diff --git a/lib/ImageTransformer.js b/lib/ImageTransformer.js
index 2f24368d133badbb520fb3396a7a92cd1f640399..3026dbef4ac11963e3400576bfc875d520aa60b8 100644
--- a/lib/ImageTransformer.js
+++ b/lib/ImageTransformer.js
@@ -55,7 +55,8 @@ var _default = new (_plugin().Transformer)({
         });
       }
 
-      asset.setStream(imagePipeline);
+      let buf = await imagePipeline.toBuffer();
+      asset.setBuffer(buf);
     }
 
     return [asset];
